modules/objdetect/include/opencv2/objdetect/aruco_board.hpp

```diff
@@ -138,6 +138,16 @@ class CV_EXPORTS_W_SIMPLE CharucoBoard : public Board {
     CV_WRAP CharucoBoard(const Size& size, float squareLength, float markerLength,
                          const Dictionary &dictionary, InputArray ids = noArray());
 
+    /** @brief set BBF chessboard pattern.
+     *
+     * BBF "black box first" creates chessboard patterns starting with a black box,
+     * otherwise it starts with a white box (WBF), see https://github.com/opencv/opencv/issues/23152.
+     *
+     * Default behavior: WBF for even row, BBF for odd row patterns.
+     */
+    CV_WRAP void setBBF(bool BBF);
+    CV_WRAP bool getBBF() const;
```

Comment:
@AleksandrPanov Did you really verify that?

Default pattern must be like this: https://github.com/opencv/opencv/blob/4.7.0/doc/pattern.png

---

We start discussion of this patch with `setLegacyPattern()` flag. No aggrement to use BBF or other unclear abbreviations (`first`, `even`, `odd` in one description is a true mess).
Description should mention changes after 4.5.5/4.6.0 releases.

Default behavior must be preserved (4.7.0).

Reply:
@alalek, I liked @stefan523's idea to give a logical name to the board design parameter. I think. that @stefan523 need choose a different name (instead of `BBF`) and improve the description in the documentation.

Pattern design was changed to implement for calibration compatibility with charuco chessboards (interactive calibration tool + kalibr). For a better understanding, I suggest using this board design parameter as a new parameter, and not just a flag for compatibility with old solutions. I could suggest the name of the parameter as `chessboardPatternCompatibility`.

I was wrong, the default behavior should be like in 4.7.0.